tests/unit/amo/reducers/test_userAbuseReports.js

```diff
@@ -0,0 +1,170 @@
+          byUserId: {
+            [userId]: { uiVisible: false },
+          },
+        });
+      });
+    });
+
+    describe('showUserAbuseReportUI', () => {
+      it('sets the uiVisible state to true', () => {
+        const userId = createUserAccountResponse().id;
+        const state = userAbuseReportsReducer(
+          initialState, showUserAbuseReportUI({ userId }));
+
+        expect(state).toMatchObject({
+          byUserId: {
+            [userId]: { uiVisible: true },
+          },
+        });
+      });
+    });
+
+    describe('sendUserAbuseReport', () => {
+      function defaultParams(params = {}) {
+        return {
+          errorHandlerId: 'some-error-handler',
+          message: 'The add-on is malware',
+          userId: createUserAccountResponse().id,
+          ...params,
+        };
+      }
+
+      it('dispatches sendUserAbuseReport', () => {
+        const params = defaultParams();
+        const action = sendUserAbuseReport(params);
+
+        expect(action.type).toEqual(SEND_USER_ABUSE_REPORT);
+        expect(action.payload).toEqual(params);
+      });
+    });
+
+    describe('loadUserAbuseReport', () => {
+      function abuseReportResponse(params = {}) {
+        return createFakeUserAbuseReport({
+          message: 'I am Groot!',
+          user: createUserAccountResponse({ id: 12 }),
+          ...params,
+        });
+      }
+
+      it('saves the abuse report response to the reducer', () => {
+        const abuseReport = abuseReportResponse();
+        const state = userAbuseReportsReducer(
+          initialState, loadUserAbuseReport({
+            message: abuseReport.message,
+            reporter: abuseReport.reporter,
+            userId: abuseReport.user.id,
+          }));
+
+        expect(state.byUserId[abuseReport.user.id].message)
+          .toEqual('I am Groot!');
+      });
+
+      it('requires a defined reporter', () => {
+        expect(() => {
+          const partialParams = abuseReportResponse();
+          delete partialParams.reporter;
+
+          loadUserAbuseReport(partialParams);
+        }).toThrow('reporter cannot be undefined');
+      });
+
+      it('allows reporter to be `null`', () => {
+        expect(() => {
+          const paramsWithNull = abuseReportResponse({ reporter: null });
+
+          loadUserAbuseReport(paramsWithNull);
+        }).not.toThrow('reporter cannot be undefined');
+      });
+
+      it('allows abuse reports for multiple users', () => {
+        const { store } = dispatchClientMetadata();
+
+        const firstReport = createFakeUserAbuseReport({
+          message: 'This user is mean',
+          reporter: null,
+          user: createUserAccountResponse({ id: 50 }),
+        });
+        const secondReport = createFakeUserAbuseReport({
+          message: 'This user is boring',
+          reporter: null,
+          user: createUserAccountResponse({ id: 51 }),
+        });
+
+        store.dispatch(loadUserAbuseReport({
```

Comment:
Why does this use a store instead of calling the reducer directly? Nothing else in this suite does that. It will execute all other reducers on `dispatch()` which seems unnecessary.

I think I asked this in a previous review but I can't find the the thread anymore.

Reply:
I think I just found it nicer than constantly returning the reducer's new state. But I can change it, no worries.